src/modules/i18n/containers/I18nProvider.js

```diff
@@ -74,26 +74,26 @@ export class I18nProvider extends React.Component<PropsType, {
     )
   }
 
-  getPredeterminedLanguage (knownLanguage: string): string {
-    if (knownLanguage) {
-      return knownLanguage
-    } else if (this.i18n.languages && this.i18n.languages.length > 0) {
+  getI18nextLanguage (): string {
+    if (this.i18n.languages && this.i18n.languages.length > 0) {
       return this.i18n.languages[0]
     } else {
       throw new Error('Failed to set language because it is currently unknown and even i18next does not know it!')
     }
   }
 
-  initLanguage (language: string) {
-    const targetLanguage = this.getPredeterminedLanguage(language)
+  initLanguage () {
+    const targetLanguage = this.getI18nextLanguage()
 
     const fonts = I18nProvider.getSelectedFonts(targetLanguage)
     this.setState({language: targetLanguage, fonts})
     this.props.setUiDirection(RTL_LANGUAGES.includes(targetLanguage) ? 'rtl' : 'ltr')
+
+    moment.locale(targetLanguage)
```

Comment:
kind of bad style if we set this globally, isn't it?

Reply:
I think this is fine because we somehow want to tell moment what language it should use by default. Correct me if I'm wrong.

moment already maintains a global language which is usually "en_US".

Yes it is bad design by moment. But I'm not sure whether we can handle it it a different was properly.
Formatting can happen in libraries which use moment. If we want them to format it for the current language we have to do it like this.

Reply:
@klinzo  any opinion on this? 

I'm thinking about removing this line and statically providing a function to create a localized moment date. Something like: `this.context.localizeDate(moment(...)).format()`

Reply:
I'm okay with setting it globally. :man_shrugging: If we don't want it, we should create an issue and deal with it later. IMO this is not something we should invest much time for right now, there are more important issues.

Reply:
Are there any potential problems setting it globally?

Reply:
No, just bad style. Which actually is a problem :D

Also it is not testable.

Reply:
Okay, than you should probably do it if it doesn't take much time and create an issue else.

Reply:
I fixed it by introducing an new context. I also fixed a lot of flow errors.

Reply:
So you can remove it?

Reply:
upsi forgot that :)